analyses/data-pre-release-qc/01-data-harmonization-qc.Rmd

```diff
@@ -0,0 +1,752 @@
+                is.na(RNA_library),
+                sample_type == "Tumor",
+                cohort != "DGD") %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+missing_ids <- setdiff(hist_ids, biospecimen_ids)
+missing_hist_ids <- setdiff(biospecimen_ids, hist_ids)
+if (length(missing_ids) > 0) {
+  output_file <- "histologies-samples-missing-in-biospecimen_id_to_bed_map.tsv"
+  missing_ids_df <- missing_biospecimen_table(hist_df, missing_ids, output_file)
+  missing_ids_df
+}
+print(paste("DNA biospecimen in histolgies missing in biospecimen_id_to_bed_map.tsv = ", length(missing_ids)))
+if (length(missing_hist_ids) > 0) {
+  output_file <- "biospecimen_id_to_bed_map-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("DNA biospecimen in biospecimen_id_to_bed_map.tsv missing in histolgies = ", length(missing_hist_ids)))
+```
+
+#### Check RNA splice events and fusions biospecimen IDs
+
+- rMATS file (splice-events-rmats.tsv.gz)
+```{r}
+# check that all biospecimen IDs are in the file
+biospecimen_ids <- data.table::fread(file.path(data_dir,
+                                               "splice-events-rmats.tsv.gz"),
+                                     data.table = FALSE) %>%
+  dplyr::pull(sample_id) %>%
+  unique()
+hist_ids <- hist_df %>%
+  dplyr::filter(experimental_strategy == "RNA-Seq",
+                sample_type == "Tumor",
+                cohort != "TCGA") %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+missing_ids <- setdiff(hist_ids, biospecimen_ids)
+missing_hist_ids <- setdiff(biospecimen_ids, hist_ids)
+if (length(missing_ids) > 0) {
+  output_file <- "histologies-samples-missing-in-splice-events-rmats.tsv"
+  missing_ids_df <- missing_biospecimen_table(hist_df, missing_ids, output_file)
+  missing_ids_df
+}
+print(paste("RNA biospecimen in histolgies missing in splice-events-rmats.tsv.gz = ", length(missing_ids)))
+if (length(missing_hist_ids) > 0) {
+  output_file <- "splice-events-rmats-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("RNA biospecimen in splice-events-rmats.tsv.gz missing in histolgies = ", length(missing_hist_ids)))
+```
+
+- AnnoFuse file (fusion-annoFuse.tsv.gz)
+```{r}
+# check that all biospecimen IDs are in the file
+biospecimen_ids <- readr::read_tsv(file.path(data_dir, "fusion-annoFuse.tsv.gz")) %>%
+  dplyr::pull(Sample) %>%
+  unique()
+hist_ids <- hist_df %>%
+  dplyr::filter(experimental_strategy == "RNA-Seq" | 
+                  experimental_strategy == "Targeted Sequencing", 
+                !is.na(RNA_library), 
+                sample_type == "Tumor",
+                !cohort %in% c("DGD", "TCGA")) %>%
```

Comment:
I am not actually sure if DGD would be expected in this file. I do not think so, as we annotated it separately with FusionAnnotator and ran the annoFuse QC on it.

Due to missing libraries, we should find alternate logic for this

```suggestion
  dplyr::filter(experimental_strategy == "RNA-Seq", 
                !cohort %in% c("DGD", "TCGA", "GTEx")) %>%
```
would this take care of it?